File: src/classes/Team.java
package classes;

import java.util.ArrayList;

public class Team {
    private String TeamName;
    private String TeamDescription;
    private ArrayList<Task> teamTasks;
    private int IDnumb;

    public Team(String name, String desc, int n){
        if (name.length() > 20){
            System.out.print("ERROR: classes.Team name too long");
        } else{
            TeamName = name;
        }
        TeamDescription = desc;
        IDnumb = n;
        teamTasks = new ArrayList<>();
    }

    public Team(String name){
        if (name == "N/A"){
            TeamName = "N/A";
        }
    }

    public String getTeamName(){
        if(TeamName == "N/A"){
            return "null";
        } else{
            return TeamName;
        }
    }

    public void assignTask(Task tk){
        teamTasks.add(tk);
    }

    

}

File: src/classes/Team.kt
//Class team written in Krotlin
package classes

import java.util.*

class Team {
    private var TeamName: String? = null
    var teamDescription: String? = null
        private set
    private var teamTasks: ArrayList<Task>? = null //array holding all tasks assigned to this team
    var iD = 0
        private set

    constructor(name: String, desc: String?, n: Int) {
        if (name.length > 20) {
            print("ERROR: classes.Team name too long")
        } else {
            TeamName = name
        }
        teamDescription = desc
        iD = n
        teamTasks = ArrayList()
    }

    //if only the name is passed to the constructor and is of value "N/A", the empty or not assigned team is created
    constructor(name: String) {
        if (name == "N/A" || name.isEmpty()) {
            TeamName = "N/A"
        }
    }

    fun getTeamName(): String? {
        return if (TeamName == "N/A") {
            "null"
        } else {
            TeamName
        }
    }

    fun assignTask(tk: Task) {
        teamTasks!!.add(tk)
    }
}